invenio/b2share/modules/b2deposit/restful.py

```diff
@@ -216,13 +222,28 @@ def get(self, oauth, domain_name, **kwargs):
         record_list = []
         start = page_offset * page_size
         stop = start + page_size
+
         for record_id in record_ids[start : stop]:
             record_details = get_record_details(record_id, curr_user_email)
             record_list.append(record_details)
-        if stop < len(record_ids):
-            record_list.append('...') # continuation indicator
-        return record_list
 
+        if len(record_list) == 0:
+            resp = jsonify({})
+        elif len(record_ids) == 1 and page_offset == 0:
+            resp  = jsonify(Deposit = record_list)
+        elif len(record_ids) == 1 and page_offset > 0:
+            resp = jsonify({})
+        elif len(record_ids) > 1 and page_offset == 0:
+            resp = jsonify(Deposits = record_list)
+        elif len(record_ids) > 1 and page_offset > 0 and len(record_list) > 0:
+            resp = jsonify(Deposits = record_list)
+        elif len(record_ids) > 1 and page_offset > 0 and len(record_list) == 0:
+            resp = jsonify({})
+
+        resp.headers['Number of Deposits'] = len(record_ids)
```

Comment:
Is it customary to return this type of information in a response header? It's a new thing for me.